src_c/sdlmain_osx.m

```diff
@@ -316,51 +307,56 @@ static void setupWindowMenu(void)
 
 static PyObject*
 _ScrapPut(PyObject *self, PyObject *args) {
-#if defined (PYGAME_MAC_SCRAP_OLD)
-	return Py_None;
-#else
-	PyObject *ret = NULL;
 	char *scrap_type;
 	char *data;
 
-	if (!PyArg_ParseTuple (args, "ss", &scrap_type, &data))
-		return Py_None;
+#if defined (PYGAME_MAC_SCRAP_OLD)
+    Py_RETURN_NONE;
+#else
+
+	if (!PyArg_ParseTuple (args, "ss", &scrap_type, &data)) {
+        Py_RETURN_NONE;
+    }
 
 	// anything else than text is not implemented
-	if (strcmp(scrap_type, PYGAME_SCRAP_TEXT))
-		return Py_None;
+	if (strcmp(scrap_type, PYGAME_SCRAP_TEXT)) {
+        Py_RETURN_NONE;
+    }
 
 	NSAutoreleasePool * pool = [[NSAutoreleasePool alloc] init];
 	NSPasteboard *pb = [NSPasteboard generalPasteboard];
 	NSString *ndata = [NSString stringWithCString:(char *)data encoding:NSUTF8StringEncoding];
 	[pb declareTypes: [NSArray arrayWithObject:NSStringPboardType] owner: pb];
 	[pb setString:ndata forType: NSStringPboardType];
 	[pool release];
-	return Py_None;
+    Py_RETURN_NONE;
 #endif
 }
 
 static PyObject*
 _ScrapSetMode(PyObject *self, PyObject *args) {
 #if defined (PYGAME_MAC_SCRAP_OLD)
-	return Py_None;
+    Py_RETURN_NONE;
 #else
 	char *mode;
-	if (!PyArg_ParseTuple (args, "s", &mode))
-		return Py_None;
-	return Py_None;
+	if (!PyArg_ParseTuple (args, "s", &mode)) {
```

Comment:
There's a Coverity issue for this bit of code if you are interested in addressing it as well.

> CID 1486568 (# 1 of 1): Identical code for different branches (IDENTICAL_BRANCHES)
> identical_branches: The same code is executed when the condition !PyArg_ParseTuple(args, "s", &mode) is true or false, because the code in the if-then branch and after the if statement is identical. Should the if statement be removed?

Reply:
Ah. I guess these failing PyArg_ParseTuple ones should be `return NULL`.

Reply:
Yes, that is what is done in the other files.

Reply:
I put `return NULL` for all the PyArg_ParseTuple failed cases.